packages/grid/x-data-grid/src/internals/hooks/features/keyboard/useGridKeyboardNavigation.ts

```diff
@@ -166,12 +168,28 @@ export const useGridKeyboardNavigation = (
   );
 
   const handleColumnHeaderKeyDown = React.useCallback<
-    GridEventListener<GridEvents.columnHeaderNavigationKeyDown>
+    GridEventListener<GridEvents.columnHeaderKeyDown>
   >(
     (params, event) => {
-      if (!params.field) {
+      const headerTitleNode = event.currentTarget.querySelector(
+        `.${gridClasses.columnHeaderTitleContainerContent}`,
+      );
+      const isFromCustomHeader =
+        !!headerTitleNode && headerTitleNode.contains(event.target as Node | null);
+      const isGridHeaderCellRoot =
+        event.target != null &&
+        (event.target as HTMLElement).classList.contains(gridClasses.columnHeader);
+
+      if (!isGridHeaderCellRoot && params.field !== GRID_CHECKBOX_SELECTION_COL_DEF.field) {
```

Comment:
I can't remember correctly why we need to ignore when the target is not the root but would be possible to simplify this condition to?

```suggestion
      if (isFromInsideContent && params.field !== GRID_CHECKBOX_SELECTION_COL_DEF.field) {
        return;
      }
```

The discussion I found https://github.com/mui/mui-x/pull/3624#discussion_r787804724

Reply:
Yes, I though I should keep it to let buttons be triggered by <kbd>Space</kbd> But they are listenig to <kbd>Enter</kbd> 
 and <kbd>Ctrl</kbd>+<kbd>Enter</kbd> so yes I'm simplifying it